labs/lab02/frontend/lib/chat_screen.dart

```
@@ -12,34 +12,94 @@ class ChatScreen extends StatefulWidget {
 }
 
 class _ChatScreenState extends State<ChatScreen> {
-  // TODO: Add TextEditingController for input
-  // TODO: Add state for messages, loading, and error
-  // TODO: Subscribe to chatService.messageStream
-  // TODO: Implement UI for sending and displaying messages
-  // TODO: Simulate chat logic for tests (current implementation is a simulation)
+  final TextEditingController _controller = TextEditingController();
+  List<String> _messages = [];
+  bool _loading = true;
+  String? _error;
+  StreamSubscription<String>? _sub;
 
   @override
   void initState() {
     super.initState();
-    // TODO: Connect to chat service and set up listeners
+    widget.chatService.connect().then((_) {
+      _sub = widget.chatService.messageStream.listen((msg) {
+        setState(() {
+          _messages.add(msg);
+        });
+      });
+      setState(() {
+        _loading = false;
+      });
+    }).catchError((err) {
+      setState(() {
+        _error = 'Connection error: $err';
+        _loading = false;
+      });
+    });
   }
 
   @override
   void dispose() {
-    // TODO: Dispose controllers and subscriptions
+    _sub?.cancel();
+    _controller.dispose();
     super.dispose();
   }
 
   void _sendMessage() async {
-    // TODO: Send message using chatService
+    final text = _controller.text;
+    if (text.isEmpty) return;
+    try {
+      await widget.chatService.sendMessage(text);
+      _controller.clear();
+    } catch (_) {}
```

The empty catch block in _sendMessage() should at least log the error